Plugin/Magento/GiftCard/GenerateGiftCardAccountsOrderPlugin.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2019 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+namespace Bolt\Boltpay\Plugin\Magento\GiftCard;
+
+use Bolt\Boltpay\Model\Payment as BoltPayment;
+use Magento\Framework\Event\Observer;
+use Magento\Sales\Model\Order;
+
+class GenerateGiftCardAccountsOrderPlugin
+{
+    public function aroundExecute(
+        $subject,
+        callable $proceed,
+        Observer $observer
+    ) {
+        $event = $observer->getEvent();
+        /** @var Order $order */
+        $order =  $event->getOrder();
+        if ($order->getPayment()->getMethod() != BoltPayment::METHOD_CODE ||
```

Comment:
Having the check here is good, but I still would like to hear about your testing here. Did you test both Bolt-enabled and Bolt-disabled paths here to make sure we didn't modify functionality for non-Bolt orders?

Reply:
@matt-thomason The \Magento\GiftCard\Observer\GenerateGiftCardAccountsOrder observer listens to sales_order_save_after event, which fires multiple times for every order. In our plugin we suppress its execution only in cases when payment method is Bolt in combination with pending_payment or canceled order statuses. Maybe it would be more clear I had written it like this:
```php
if ($order->getPayment()->getMethod() === BoltPayment::METHOD_CODE && 
    in_array($order->getStatus(), [Order::STATE_PENDING_PAYMENT,Order::STATE_CANCELED])
) {
    return;
}
return $proceed($observer);
```
Anyway, I did not test with Bolt disabled.

Reply:
Updated as above.

Reply:
Thanks @danac-gs - I did not think the previous code was unclear though. I just wanted to confirm what you manually tested. Could you do a manual test with a non-Bolt payment method and confirm back here? Testing with Bolt disabled probably is overkill, I agree.

Reply:
@matt-thomason I've checked, it works. Bought a GC with check / money order payment and it went through. Please re-approve.